pkg/binding/binding.go

```diff
@@ -101,58 +93,23 @@ func (c *Controller) updateOrCreateBinding(ctx context.Context, bdg *v1alpha1.Bi
 	}
 	bdg.SetOwnerReferences([]metav1.OwnerReference{ownerReference})
 
-	// update or create binding
-	unstructuredBinding, err := bindingToUnstructuredObject(bdg)
-	if err != nil {
-		return fmt.Errorf("failed to convert Binding to Unstructured: %w", err)
-	}
-
 	logger := klog.FromContext(ctx)
+	bdgEcho, err := c.controlClient.Bindings().Update(ctx, bdg, metav1.UpdateOptions{FieldManager: controllerName})
 
```

Comment:
We did not use a field manager before, right ? This sounds like a good idea, just wondering if this is something that could create issues with existing bindings (if any) and if yes, if users should be informed about this.

Reply:
As far as I know, we do not support in-place upgrades. If we did then I think it would take some more thorough review, documentation, and testing. And probably some real development work, I expect that issues would turn up.

Reply:
I agree with @MikeSpreitzer - I think we have an open issue to support upgrades, I also think it is very important, however I believe we have several other "gaps"  we need to close to support it besides this one.